src/index.ts

```diff
@@ -6,4 +6,16 @@ declare module '@webcomponents/custom-elements';
   }
 
   require('./init-app.tsx');
+
+  // Initialize the command queue in case analytics.js hasn't loaded yet.
+  window.ga = window.ga || ((...args) => (ga.q = ga.q || []).push(args));
```

Comment:
I don’t if/how this gets transpiled by TypeScript, but have you made sure that this actually works? (You can just check the DevTools Network tab to see if a beacon goes out.)

I’m asking because of https://github.com/v8/v8.dev/blob/d6805e0423c7301d019a5fba3c19bc1c8c304cda/src/_js/main.mjs#L45-L61, which seems like a similar scenario that definitely surprised me at the time.

Reply:
I got this from [@philipwalton’s blog post](https://philipwalton.com/articles/the-google-analytics-setup-i-use-on-every-site-i-build/).

I didn’t realize this wasn’t the official snippet. But you are right, the param spread might actually break in some browsers. I’ll fix that. Thanks!

Reply:
Note that the issue I ran into was not even related to browser support. Glad to see that `ga.js` apparently doesn’t care about whether something is an `arguments` object or an actual array (unlike `gtag`)!